From: Egbert Eich <eich@suse.de>
Date: Tue Jul 5 15:18:26 2016 +0200
Subject: [PATCH]shadow: Calulate the shadow buffer size correctly
Patch-mainline: to be upstreamed
Git-commit: e9ff02d4037cd4c693fdef3dacc1bb4fc282455c
References: boo#987670
Signed-off-by: Egbert Eich <eich@suse.com>

The 'width' used for the calculation of the shadow buffer is
not the width of the screen but the framebuffer stride in of
the shadow buffer in pScrn->displayWidth. This fixes a cut-off
lower border.

Signed-off-by: Egbert Eich <eich@suse.de>
Found-by: Oliver Schweigert <oliver.schweikert@horiba.com>
---
 src/mga_driver.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: xf86-video-mga-2.1.0/src/mga_driver.c
===================================================================
--- xf86-video-mga-2.1.0.orig/src/mga_driver.c
+++ xf86-video-mga-2.1.0/src/mga_driver.c
@@ -3323,7 +3323,7 @@ MGAScreenInit(SCREEN_INIT_ARGS_DECL)
     }
 
     if(pMga->ShadowFB) {
-	pMga->ShadowPtr = (unsigned char *)xnfcalloc(1, width * height *
+	pMga->ShadowPtr = (unsigned char *)xnfcalloc(1, pScrn->displayWidth * height *
                                                      (pScrn->bitsPerPixel >> 3));
         FBStart = pMga->ShadowPtr;
     } else {
